app/controllers/search_controller.rb

```diff
@@ -3,10 +3,19 @@ def index
 
   end
 
-  def show
-    # replace non-ASCII characters with ASCII and non-word characters with spaces for API call
+  # def show
+  #   # replace non-ASCII characters with ASCII and non-word characters with spaces for API call
+  #   sanitized_search = I18n.transliterate(params[:q]).gsub(/(\W)/, " ")
+  #   @artist = Search.for(sanitized_search).first
+  #   @related_artists = @artist.related_artists.sort_by {|artist| artist.followers["total"]}.reverse
+  # end
+
+  def data
     sanitized_search = I18n.transliterate(params[:q]).gsub(/(\W)/, " ")
-    @artist = Search.for(sanitized_search).first
-    @related_artists = @artist.related_artists.sort_by {|artist| artist.followers["total"]}.reverse
+    @artist = Search.for(sanitized_search)[0]
+    children_artists = @artist.related_artists
+    grandchildren_artists = children_artists.map do |artist|
+      artist.related_artists.delete_if {|artist| artist.name == @artist.name}
```

Comment:
Shadowing outer local variable - `artist`.<br>Space between { and | missing.<br>Space missing inside }.